mapadroid/worker/WorkerBase.py

```diff
@@ -1181,3 +1182,41 @@ def _get_screen_size(self):
         self._resocalc.get_x_y_ratio(
             self, self._screen_x, self._screen_y, x_offset, y_offset)
 
+    def _check_data_distance(self, data):
+        mode = self._mapping_manager.routemanager_get_mode(self._routemanager_name)
+        if mode in ["mon_mitm", "iv_mitm"]:
+            data_to_check = "wild_pokemon"
+        else:
+            data_to_check = "forts"
```

Comment:
for stops and gyms, you can legitimately get information that is further away than max radius, - has it been looked into that this edge case does not cause "too far away" timeouts? 
Example of such a situation:
![image](https://user-images.githubusercontent.com/5710881/76419946-8de98f80-63a1-11ea-98d9-8dcab34cfbb1.png)

The left circle has a row of gyms and stops that are all outside the radius, but still "in range" because of hos s2 cells work.
An alternative way to figure out this max distance is to do the calculation we do when visualizing the coverage in raid mode - and then use the distance from centre to the top left corner etc

Reply:
Using the s2Helper stuff looks promising per se - just using the center of all `middle_of_cell`s would probably yield a more precise result, but I'm just not quite sure how to interpret the GMO data. I just looked at a couple of data samples and there always seem to be cells with nothing but their ID in the data:

`{"id":1234,"current_timestamp":9876,"forts":[],"wild_pokemon":[],"nearby_pokemon":[],"catchable_pokemon":[],"decimated_spawnpoints":[],"spawnpoints":[]}`

Would I just use those cells that contain data relevant to the scan mode?
For a real improvement, someone who knows more about those behaviours should probably look at this and tell me how to interpret the GMO data for this purpose, or just quickly rework it on their own.

Reply:
The amount of empty cells is related to the distance to the player. A lot more cells are sent to the client than those populated with mons since forts, weather etc also play a role. For mon-scanning, I'd ignore those accordingly.

Reply:
Finally got around to implement this with the latest commit. Basically just stole some code from here:
https://github.com/Map-A-Droid/MAD/blob/936ffb821ccd80d8382227904f11a3af29eb7b5a/mapadroid/db/DbPogoProtoSubmit.py#L629-L635.

I now used `get_position_from_cell` instead of `middle_of_cell` because it was used in the snippet above :D ... it seems to do exactly the same, just that the first one also returns a altitude of 0 that we don't need here?